src/PrestaShopBundle/Form/Admin/Improve/International/Translations/TranslationsSettingsFormHandler.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Form\Admin\Improve\International\Translations;
+
+use PrestaShop\PrestaShop\Core\Form\FormHandlerInterface;
+use PrestaShop\PrestaShop\Core\Hook\HookDispatcherInterface;
+use Symfony\Component\Form\FormBuilderInterface;
+
+final class TranslationsSettingsFormHandler implements FormHandlerInterface
+{
+    /**
+     * @var FormBuilderInterface the form builder
+     */
+    protected $formBuilder;
+
+    /**
+     * @var HookDispatcherInterface the event dispatcher
+     */
+    protected $hookDispatcher;
+
+    /**
+     * @var string the hook name to be dispatched
+     */
+    protected $hookName;
+
+    /**
+     * @var array the list of Form Types
+     */
+    protected $formTypes;
+
+    /**
+     * @param FormBuilderInterface $formBuilder
+     * @param HookDispatcherInterface $hookDispatcher
+     * @param array $formTypes
+     * @param string $hookName
+     */
+    public function __construct(
+        FormBuilderInterface $formBuilder,
+        HookDispatcherInterface $hookDispatcher,
+        array $formTypes,
+        $hookName
+    ) {
+        $this->formBuilder = $formBuilder;
+        $this->hookDispatcher = $hookDispatcher;
+        $this->formTypes = $formTypes;
+        $this->hookName = $hookName;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getForm()
+    {
+        foreach ($this->formTypes as $formName => $formType) {
+            $this->formBuilder->add($formName, $formType);
+        }
+
+        $this->hookDispatcher->dispatchWithParameters(
+            "action{$this->hookName}Form",
+            [
+                'form_builder' => $this->formBuilder,
```

Comment:
btw @mickaelandrieu this proves the point that you wanted to test, that passing hook params by reference is not needed when they are objects :)

Reply:
ok :+1:  good to know, I'll clean the others calls

Reply:
will do the cs fix :)

Reply:
fixed